resources/js/components/SignUpForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation } from "@tanstack/react-query";
+import type { SubmitHandler } from "react-hook-form";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { MailIcon } from "@/icons/MailIcon";
+import { UserIcon } from "@/icons/UserIcon";
+import { signUpUser } from "@/services/authService";
+import { Button } from "./Button";
+import Input from "./Input";
+import PasswordInput from "./PasswordInput";
+
+const SignUpForm = () => {
+  const userSchema = z
+    .object({
+      mail: z.string().email(),
+      username: z.string().min(1, "Field required!"),
```

Comment:
I'm not sure, but I belive there is a required function, so you don't need to ask for min 1. If I'm right, then it probably also has a default descriptive message, so you won't need to override it

Reply:
I took it and it didn't make de required validation